### PR TITLE
Mvp

### PR DESCRIPTION
**Pull request by spacifici**

### CODE REVIEW

File: app/src/main/java/org/mozilla/reference/browser/ext/Context.kt [hunk 1]
```diff
@@ -78,4 +78,21 @@ fun Context.preferences() = PreferenceHelper.getInstance(this)
...
+ * @see android.content.res.Resources.getQuantityString
+ */
+@Suppress("SpreadOperator")
+fun Context.getQuantityString(
+    @PluralsRes pluralId: Int,
+    quantity: Int,
+    vararg formatArgs: Any
+): String {
+    return resources.getQuantityString(pluralId, quantity, *formatArgs)
+}
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/ext/Context.kt]: can be converted to an equal sign

File: app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryFragment.kt [hunk 2]
```diff
@@ -9,6 +9,7 @@ import android.view.MenuItem
 import android.view.View
 import android.view.ViewGroup
 import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryFragment.kt]: Missing license

File: app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryFragment.kt [hunk 3]
```diff
@@ -110,6 +111,10 @@ class HistoryFragment : Fragment(), UserInteractionHandler {
                 create()
                 show()
             }
+            alertDialog.getButton(AlertDialog.BUTTON_POSITIVE)
+                .setTextColor(ContextCompat.getColor(it, android.R.color.white))
+            alertDialog.getButton(AlertDialog.BUTTON_NEGATIVE)
+                .setTextColor(ContextCompat.getColor(it, android.R.color.white))
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryFragment.kt]: This should be made with theaming

File: app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryView.kt [hunk 4]
```diff
@@ -47,24 +52,28 @@ class HistoryView(
         view.toolbar.inflateMenu(layout)
     }
 
+    @SuppressLint("PrivateResource")
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/library/history/ui/HistoryView.kt]: Remove this and copy the resource in the project as suggested by the lint

File: app/src/main/java/org/mozilla/reference/browser/settings/SettingsActivity.kt [hunk 5]
```diff
@@ -9,12 +9,17 @@ import androidx.appcompat.app.AppCompatActivity
...
+import org.mozilla.reference.browser.R.color.navigationBarColor
+import org.mozilla.reference.browser.R.color.statusBarColor
+import org.mozilla.reference.browser.ext.setWindowTheme
 
 class SettingsActivity : AppCompatActivity(), SettingsFragment.ActionBarUpdater {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        setWindowTheme(statusBarColor, navigationBarColor)
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/settings/SettingsActivity.kt]: This should also be made with theming and not in code

File: app/src/main/java/org/mozilla/reference/browser/utils/ClipboardHandler.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+private const val MIME_TYPE_TEXT_PLAIN = "text/plain"
+private const val MIME_TYPE_TEXT_HTML = "text/html"
+
+/**
+ * A clipboard utility class that allows copying and pasting links/text to & from the clipboard
+ */
+class ClipboardHandler(context: Context) {
+    private val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+
+    var text: String?
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/utils/ClipboardHandler.kt]: Documentation

File: app/src/main/java/org/mozilla/reference/browser/utils/ClipboardHandler.kt [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+            ) {
+                return clipboard.firstPrimaryClipItem?.text.toString()
+            }
+            return null
+        }
+        set(value) {
+            clipboard.primaryClip = ClipData.newPlainText("Text", value)
+        }
+
+    val url: String?
```
Comment (spacifici) [inline app/src/main/java/org/mozilla/reference/browser/utils/ClipboardHandler.kt]: Documentation

File: app/src/main/res/values-night/colors.xml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (spacifici) [inline app/src/main/res/values-night/colors.xml]: License

File: app/src/main/res/values-night/themes.xml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (spacifici) [inline app/src/main/res/values-night/themes.xml]: License

File: app/src/main/res/values-notnight/colors.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (spacifici) [inline app/src/main/res/values-notnight/colors.xml]: License

File: app/src/main/res/values-notnight/themes.xml [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (spacifici) [inline app/src/main/res/values-notnight/themes.xml]: License

File: app/src/test/java/org/mozilla/reference/browser/FreshTabIntegrationTest.kt [hunk 12]
```diff
@@ -1,30 +1,26 @@
-package com.cliqz.browser.freshtab
+package org.mozilla.reference.browser
```
Comment (spacifici) [inline app/src/test/java/org/mozilla/reference/browser/FreshTabIntegrationTest.kt]: License

File: app/lint.xml [hunk 13]
```diff
@@ -11,5 +11,12 @@
     </issue>
     <issue id="UnusedResources">
         <ignore path="src/main/res/values/dimens.xml" />
+        <ignore path="src/main/res/values/colors.xml" />
+        <ignore path="src/main/res/values-notnight/colors.xml" />
+        <ignore path="src/main/res/values-night/colors.xml" />
+    </issue>
+    <issue id="MissingDefaultResource">
+        <ignore path="src/main/res/values-notnight/colors.xml" />
+        <ignore path="src/main/res/values-night/colors.xml" />
```
[the remark this answers is not in the document]
Comment (jayeshsolanki93) [reply app/lint.xml]: There are some theme-specific resources in those files. If this is not ignored, I would have to duplicate values in the base folder.

File: app/src/main/java/org/mozilla/reference/browser/settings/SettingsActivity.kt (hunk 5 above)
Comment (jayeshsolanki93) [reply app/src/main/java/org/mozilla/reference/browser/settings/SettingsActivity.kt]: I have through the theme but the status bar and navigation bar buttons/text do not change the colour to black on white theme. This image has the status bar and navigation bar to it.



Screenshot_1580311432


I can do this to fix this but this required min API 27 and 23 resp.
```
        <item name="android:windowLightNavigationBar">true</item>
        <item name="android:windowLightStatusBar">true</item>
```

File: app/src/main/java/org/mozilla/reference/browser/ext/Context.kt [hunk 14]
```diff
@@ -78,4 +78,21 @@ fun Context.preferences() = PreferenceHelper.getInstance(this)
...
+
+/**
+ * Shortcut to get quantity aware strings from the context.
+ *
+ * @param pluralId the plural id
+ * @param quantity the quantity you want the string for
+ * @param formatArgs the arguments used for substitution
+ * @see android.content.res.Resources.getQuantityString
+ */
+@Suppress("SpreadOperator")
```
[the remark this answers is not in the document]
Comment (jayeshsolanki93) [reply app/src/main/java/org/mozilla/reference/browser/ext/Context.kt]: I have used a spread operator on the next line and detekt complains about it.

**Pull request merged**